src/core/processing-queue.js

```diff
@@ -182,29 +177,46 @@ function done() {
 		if ( config.testId && config.testId.length ) {
 			throw new Error( `No tests matched the testId "${config.testId}".` );
 		}
-
 		throw new Error( "No tests were run." );
+	}
+}
+
+// Clear own storage items when tests completes
+function cleanStorage() {
+	const storage = config.storage;
+	if ( storage && config.stats.bad === 0 ) {
+		for ( let i = storage.length - 1; i >= 0; i-- ) {
+			const key = storage.key( i );
 
+			if ( key.indexOf( "qunit-test-" ) === 0 ) {
+				storage.removeItem( key );
+			}
+		}
+	}
+}
+
+/**
+ * This function is called when the ProcessingQueue is done processing all
+ * items. It handles emitting the final run events.
+ */
+function done() {
+	ProcessingQueue.finished = true;
+
+	try {
+		throwErrorIfNoTestExecuted();
+	} catch ( err ) {
+		throw err;
```

Comment:
I don't understand why you're catching the error just to re-throw it here? Maybe a comment about why this is needed

Reply:
I don't understand what I was thinking either, totally unnecessary :)